content/blog/updating-to-cpp11/index.Rmd

```diff
@@ -0,0 +1,90 @@
+---
+output: hugodown::hugo_document
+
+slug: updating-to-cpp11
+title: Pathway to success - updating your package to cpp11
+date: 2021-09-10
+author: Shelby Bearrows
+description: >
+    The cpp11 summer intern reviews the process they used to convert readxl to using 
+    cpp11.
+
+photo:
+  url: https://unsplash.com/photos/9MMd2uRpfvc
+  author: John Salzarulo
+
+# one of: "deep-dive", "learn", "package", "programming", or "other"
+categories: [learn] 
+tags: [cpp11, internship]
+---
+
+<!--
+TODO:
+* [x] Look over / edit the post's title in the yaml
+* [x] Edit (or delete) the description; note this appears in the Twitter card
+* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+Over the summer I had the pleasure of working with Jim Hester on the [cpp11 package](https://cpp11.r-lib.org/) as a tidyverse summer intern.
+The cpp11 package is a header-only R package that helps R package developers handle R objects with C++ code.
+Its goals and syntax are similar to the excellent Rcpp package.
+During most of my internship, I worked on triaging issues, fixing bugs, and adding new features to cpp11.
+Near the end of the summer, I got to work with Jenny Bryan on converting the [readxl](https://github.com/tidyverse/readxl/pull/659) package from using Rcpp to using cpp11.
```

Comment:
I included the readxl PR here!

Reply:
Aha, I see. I think this is a case where it's too easy for a reader to mis-appreciate what that hyperlink goes to. I definitely assumed it went to the readxl pkgdown site (reading quickly). I think the link to the readxl PR should explicitly labelled as such in the prose. I think the two PR hyperlinks (readxl and readr) should appear together, for maximum impact.

Here's a good article on how people process hyperlinks and the implications for usability:

https://heyoka.medium.com/dont-use-click-here-f32f445d1021

Reply:
Thanks, I'll read over this and make adjustments!